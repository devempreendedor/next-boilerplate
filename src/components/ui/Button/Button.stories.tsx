import { Button } from '.'
import { ButtonProps } from './Button'

export default {
  title: 'Example/Button',
  component: Button,
  argTypes: {
    variant: { control: 'color' },
  },
}

const Template = ({ children, ...args }: ButtonProps) => (
  <Button {...args}>{children}</Button>
)

export const Primary = Template.bind({})
Primary.args = {
  children: 'Button',
}

export const Secondary = Template.bind({})
Secondary.args = {
  children: 'Button',
  variant: 'secondary',
}
