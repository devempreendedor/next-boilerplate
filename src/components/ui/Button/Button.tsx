import React, { ReactNode } from 'react'
import { cls } from '@utils/helpers'

export interface ButtonProps {
  children: ReactNode
  type?: 'button' | 'submit' | 'reset'
  className?: string
  variant?: 'primary' | 'secondary' | 'danger'
  size?: 'small' | 'medium' | 'large'
  disabled?: boolean
  onClick?: () => void
}

const classes = {
  base: 'focus:outline-none transition ease-in-out duration-300 rounded-lg',
  disabled: 'opacity-50 cursor-not-allowed',
  size: {
    small: 'px-2 py-1 text-sm',
    medium: 'px-4 py-2',
    large: 'px-8 py-3 text-lg',
  },
  variant: {
    primary:
      'bg-indigo-500 hover:bg-indigo-800 focus:ring-2 focus:ring-indigo-500 focus:ring-opacity-50 text-white',
    secondary:
      'bg-gray-200 hover:bg-gray-800 focus:ring-2 focus:ring-gray-500 focus:ring-opacity-50 text-gray-900 hover:text-white',
    danger:
      'bg-red-500 hover:bg-red-800 focus:ring-2 focus:ring-red-500 focus:ring-opacity-50 text-white',
  },
}

const Button = ({
  children,
  type = 'button',
  className,
  variant = 'primary',
  size = 'medium',
  disabled = false,
  ...props
}: ButtonProps) => {
  return (
    <button
      type={type}
      disabled={disabled}
      className={cls(`
    ${classes.base}
    ${classes.size[size]}
    ${classes.variant[variant]}
    ${disabled && classes.disabled}
    ${className}
`)}
      {...props}
    >
      <span>{children}</span>
    </button>
  )
}

export default Button
