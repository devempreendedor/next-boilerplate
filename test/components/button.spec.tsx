import { render, screen } from '../test-utils'
import { Button } from '@components/ui/Button'

describe('Button Component', () => {
  it('Should render the button', () => {
    render(<Button>Button</Button>)

    const buttonText = screen.getByText(/Button/i)

    expect(buttonText).toBeInTheDocument()
  })
})
