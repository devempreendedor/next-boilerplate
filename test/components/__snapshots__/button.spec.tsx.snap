// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render the button 1`] = `
<span>
  Button
</span>
`;
